src/test/features/claimant-response/helpers/settlementHelper.ts

```diff
@@ -0,0 +1,117 @@
+    })
+
+    it('should throw error when settlement agreement is undefined', () => {
+      draft = new DraftClaimantResponse().deserialize({ ...sampleClaimantResponseDraftObj, ...{ settlementAgreement: undefined } })
+      expect(() => prepareSettlement(claim, draft)).to.throw(Error, 'SettlementAgreement should be signed by claimant')
+    })
+
+    it('should throw error when settlement is not signed', () => {
+      draft = new DraftClaimantResponse().deserialize({ ...sampleClaimantResponseDraftObj, ...{ settlementAgreement: { signed: false } } })
+      expect(() => prepareSettlement(claim, draft)).to.throw(Error, 'SettlementAgreement should be signed by claimant')
+    })
+  })
+
+  describe('getRepaymentPlanOrigin', () => {
+    it('should return DEFENDANT if offer accepted was made by defendant', () => {
+      const defendantStatement = new PartyStatement(StatementType.OFFER.value, 'DEFENDANT', new Offer('contents', MomentFactory.currentDate()))
+      const partyStatements: PartyStatement[] = [defendantStatement, acceptOffer()]
+      const settlement: Settlement = new Settlement(partyStatements)
+
+      const repaymentPlanOrigin: string = getRepaymentPlanOrigin(settlement)
+      expect(repaymentPlanOrigin).is.not.undefined
+      expect(repaymentPlanOrigin).to.be.eql('DEFENDANT')
+    })
+
+    it('should return CLAIMANT if offer accepted was made by claimant', () => {
+      const defendantStatement = new PartyStatement(StatementType.OFFER.value, 'CLAIMANT', new Offer('contents', MomentFactory.currentDate()))
+      const partyStatements: PartyStatement[] = [defendantStatement, acceptOffer()]
+      const settlement: Settlement = new Settlement(partyStatements)
+
+      const repaymentPlanOrigin: string = getRepaymentPlanOrigin(settlement)
+      expect(repaymentPlanOrigin).is.not.undefined
+      expect(repaymentPlanOrigin).to.be.eql('CLAIMANT')
+    })
+
+    it('should return COURT if offer accepted was made by court', () => {
+      const defendantStatement = new PartyStatement(StatementType.OFFER.value, 'COURT', new Offer('contents', MomentFactory.currentDate()))
+      const partyStatements: PartyStatement[] = [defendantStatement, acceptOffer()]
+      const settlement: Settlement = new Settlement(partyStatements)
+
+      const repaymentPlanOrigin: string = getRepaymentPlanOrigin(settlement)
+      expect(repaymentPlanOrigin).is.not.undefined
+      expect(repaymentPlanOrigin).to.be.eql('COURT')
+    })
+
+    it('should throw error if settlement is undefined', () => {
+      expect(() => getRepaymentPlanOrigin(undefined)).to.throw(Error, 'settlement must not be null')
+    })
+
+    it('should throw error if partyStatement is undefined', () => {
+      expect(() => getRepaymentPlanOrigin(new Settlement([undefined, acceptOffer()]))).to.throw(Error, 'partyStatement must not be null')
+    })
+  })
+
+  describe('acceptOffer', () => {
+    it('should generate acceptation party statement', () => {
```

Comment:
can you put a check on the generated off too please?

Reply:
we don't have `generated` flag any more after the new design

Reply:
for acceptation of offer, we don't populate offer object. it is actually populated in previous party statement, accept offer has only:
```
 new PartyStatement(StatementType.ACCEPTATION.value, MadeBy.CLAIMANT.value)
```

Reply:
can you add that in the defendant section then please?

Reply:
done